### ISSUE TITLE
Filter items with pockets by pocket's size, len, etc.

### ISSUE THREAD

**Issue by Brambor** [CleverRaven/Cataclysm-DDA#74635]
### Is your feature request related to a problem? Please describe.

It is very tedious whether I have an item with a pocket that can contain a 78 cm long item (pickaxe). Or if I can craft such an item.

### Solution you would like.

Add a filter such as `l:X` (l for length) to filter items with a pocket that can contain items that are X cm long.

This isn't perfect by itself, but it would filter out all the clothing etc. on the tile that definitely cannot contain items with selected lengths.

### Describe alternatives you have considered.

Currently one of the ways is to read this menu in the item description and search for every item there, whether I have them in my base and then search again, if I can craft this item:
![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/13402666/54ed8028-b22f-4f30-a879-1b161608331e)

This method is feasible for plank:
![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/13402666/cc5cc09d-0986-4256-ab5c-45b2f31b5a93)

Otherwise, if the menu text (Can be stored in) would highlight items that are nearby, I would only need to search the crafting menu, if I don't have any item nearby.

**Comment by silvadomi** [comment]
Such a filtering option seems useful indeed. Or, maybe even more direct, filter by items that have pockets which can hold a specific item? Like a condition such as `f:pickaxe` that would show all items that a pickaxe can fit into.

**Comment by Brambor** [comment]
> Such a filtering option seems useful indeed. Or, maybe even more direct, filter by items that have pockets which can hold a specific item? Like a condition such as `f:pickaxe` that would show all items that a pickaxe can fit into.

Yes, this is more straightforward. This also avoids the problem of putting `cm` vs `inches` into the filter.
